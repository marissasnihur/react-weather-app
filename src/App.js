/*
This is the root level component that brings in and renders the entire React application. It contains
an internal state, high level styling, and the declartion of the application's' hierarchy.

changeDisplayCity:
  When this function is triggered, it will perform 2 network calls to retrieve data. For each
  set of returning data, we will set update this component's state with the new information which
  in turn will trigger a re-render of the UI.

componentDidMount:
  This is one of many React component's lifecycle method. After the render cycle of a component, we
  want to fetch the weather information for the DEFAULT_CITY and display it after the page is fully
  initialized.

onCitySelectionChanged:
  This function is being passed down to <CitySelection> as a prop. Upon the change of the dropdown
  selection, the component will trigger this function which in turn allow us to relay the trigger
  to `changeDisplayCity`. Obtaining the user selected value out of the dropdown is critical, so we
  will retrieve it by calling `event.target.value` which referenced to the event caller, in this
  case the dropdown menu.

render:
  This method in a React component is responsible for the rendering. Here we can describe what is
  being rendered, and how information is being used.
*/

import React, { Component } from 'react';

import WeatherCard from './containers/weatherCard';
import CitySelection from './components/citySelection/citySelection';

import { getWeatherByCity, getForecastByCity } from './utils/network';

import './App.scss';

const DEFAULT_CITY = 'edmonton';

class App extends Component {
  state = {
    city: '',
    temperature: '',
    currentCondition: '',
    minTemp: '',
    maxTemp: '',
  };

  changeDisplayCity(city) {
    getWeatherByCity(city).then((response) => {
      console.log(response)
      this.setState({
        city: response.data[0].city,
        temperature: Math.round(response.data[0].temperature),
        currentCondition: response.data[0].currentcondition,
        cityImage: response.data[0].photo,
        minTemp: response.data[0].mintemp,
        maxTemp: response.data[0].maxtemp
      })
    });

//     getForecastByCity(city).then((response) => {
//       this.setState({
//         forecast: response.data.list
//       })
//     });
   }

  componentDidMount() {
    this.changeDisplayCity(DEFAULT_CITY);
  }

  onCitySelectionChanged(event) {
    this.changeDisplayCity(event.target.value);
  }

  render() {
    return (
      <div className="App">
        <WeatherCard
          city={this.state.city}
          temperature={this.state.temperature}
          currentCondition={this.state.currentCondition}
          cityImage={this.state.cityImage}
          minTemp={this.state.minTemp}
          maxTemp={this.state.maxTemp}>
        </WeatherCard>
      </div>
    );
  }
}

export default App;
