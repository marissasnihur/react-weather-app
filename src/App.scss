.card {
  width: 300px;
  height: 450px;
  margin: 1em auto 0;
  position: relative;
  cursor: pointer;
}

.cardUpperPortion,
.cardBottomPortion {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 4px;
}

header {
  width: 300px;
  margin: 4em auto 0;
  text-align: center;
  display: flex;
  justify-content: center;  
}

