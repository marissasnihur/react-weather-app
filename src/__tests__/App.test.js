import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from '../App';

// See more at https://reactjs.org/docs/shallow-renderer.html

describe('Main App component', () => {
  it('should render without crashing', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<App />);
    const result = renderer.getRenderOutput();  
    const div = document.createElement('div');
    ReactDOM.render(result, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<App />);
    const result = renderer.getRenderOutput();
  
    const div = document.createElement('div');
    ReactDOM.render(result, div);
    expect(div.innerHTML).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});




