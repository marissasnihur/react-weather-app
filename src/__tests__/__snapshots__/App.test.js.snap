// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App component should match snapshot 1`] = `"<div class=\\"App\\"><header><label for=\\"citySelection\\">Select a city:</label><div class=\\"selectDropdown\\"><select id=\\"citySelection\\"><option value=\\"edmonton\\">Edmonton</option><option value=\\"vancouver\\">Vancouver</option><option value=\\"calgary\\">Calgary</option><option value=\\"montreal\\">Montreal</option><option value=\\"toronto\\">Toronto</option></select></div></header><div class=\\"card \\"><div class=\\"cardUpperPortion\\" style=\\"background-image: url(undefined);\\"><div class=\\"cardInfo\\"><div class=\\"cardTemperature\\"></div><div class=\\"currentCondition\\"></div></div><h2 class=\\"cityNameTitle\\"></h2></div><div class=\\"cardBottomPortion\\"><div class=\\"cardDetail\\"><ul class=\\"forecastsWrapper\\"></ul></div></div></div></div>"`;
